backend/src/main/resources/db/changelog/db.changelog-master.yaml

```diff
@@ -1079,3 +1079,88 @@ databaseChangeLog:
+            rollback:
+              sql: |
+                DROP TYPE ${database.defaultSchemaName}.TEST_CORRECTION_STATUS;
+        - addColumn:
+            tableName: test_order
+            columns:
+              - column:
+                  name: correction_status
+                  type: ${database.defaultSchemaName}.test_correction_status
+                  remarks: Enumeration of type of correction. This column is used for non-delete corrections.
+                  defaultValue: 'ORIGINAL'
+                  value: 'ORIGINAL'
+                  constraints:
+                    nullable: false
+              - column:
+                  name: reason_for_correction
+                  type: text
+                  remarks: User supplied comment when a correction happens.
+        - addColumn:
+            tableName: test_event
+            columns:
+              - column:
+                  name: patient_answers_data
+                  type: jsonb
+                  remarks: The patient's demographics and address at the time the test was administered.
+              - column:
+                  name: date_tested_backdate
+                  type: DATETIME
+                  remarks: Timestamp to store a user correction to when the test was ordered. Copied from TestOrder. Null if not backdated.
+              - column:
+                  name: test_order_id
+                  type: *idtype
+                  remarks: Reference to original test_order internal_id.
+              - column:
+                  name: correction_status
+                  type: ${database.defaultSchemaName}.test_correction_status
+                  remarks: Enumeration of type of correction.
+                  defaultValue: 'ORIGINAL'
+                  value: 'ORIGINAL'
+                  constraints:
+                    nullable: false
+              - column:
+                  name: prior_corrected_test_event_id
+                  type: *idtype
+                  remarks: If this is a correction, this is the id the prior test_event being corrected.
+              - column:
+                  name: reason_for_correction
+                  type: text
+                  remarks: User supplied comment when a correction happens.
+        - sql:
+            remarks: Copy the existing test_order_id from test_order.
+            sql: |
+              UPDATE ${database.defaultSchemaName}.test_event TestEvent
+              SET test_order_id = TestOrder.internal_id,
+                  patient_answers_data = PatientAnswers.ask_on_entry,
+                  date_tested_backdate = TestOrder.date_tested_backdate
+              FROM ${database.defaultSchemaName}.test_order TestOrder
+                   INNER JOIN ${database.defaultSchemaName}.patient_answers PatientAnswers
+                   ON TestOrder.patient_answers_id = PatientAnswers.internal_id
+              WHERE TestOrder.test_event_id = TestEvent.internal_id;
+        - addForeignKeyConstraint:
```

Comment:
These should be added at creation time—foreign keys can be nullable, and if there's to be a failure, it should be on the update statement, not after the update completes and the foreign key isn't able to be applied.